### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by tjprescott**
Clean up grammar and clarify the ARM review process, as I've dealt with two situations just today where dataplane PRs were flagged for ARM review because of the way the ARM review checklist was worded. 

I don't know how automation might be affected by changing the text, since the bot does automatically add tags based on certain boxes being checked.

### CODE REVIEW

**Comment by akning-ms** [comment]
@josefree  @zhenglaizhang  to review the change

**Comment by zhenglaizhang** [comment]
@ruowan to see if any impact to workflow bot

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -29,24 +29,32 @@ Please ensure to add changelog with this PR by answering the following questions
...
-### ARM API Review Checklist
-- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
+### ARM API Review Checklist 
+
+> **Applicability**: :warning: 
+>
+> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
+> - Change to data plane APIs 
+> - Adding new properties 
+> - All removals
```
Comment (tjprescott) [inline .github/PULL_REQUEST_TEMPLATE.md:39]: @zhenglaizhang I copied this phrasing over, but to me "all removals" is very vague. Does this mean property and API removals?

**Review by josefree — APPROVED** [review]
thanks, I am okay with the updates

**Review by ruowan — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (zhenglaizhang) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: yes, remove doesn't require ARM review.

**Pull request merged**